### DISCUSSION TITLE
Starting MQTT on Neuron fails | MQTT Conversion

### DISCUSSION THREAD

**Discussion by YamatoYamm** [emqx/emqx#12318] [Q&A] [answered]
Hello !

I have 2 problems and I hope that someone could help me.

First problem is MQTT in Neuron, I followed steps from https://www.emqx.com/en/blog/bridging-opc-ua-data-to-mqtt-for-iiot but when I try to run MQTT even if it's ready, I receive Unknown error (1007) and I don't receive any message in MQTTX. I've installed everything and I followed all steps. 

Another question is: Can I have OPC UA and MQTT complemented ? For example I need to have in my project OPC UA and MQTT but I don't want to convert OPC UA to MQTT, I need both same time. MQTT OSI level 4 and OPC UA higher level OSI.

**Comment by ieQu1** [comment]
Hello,

Was it meant to be asked here? https://github.com/emqx/neuron/discussions

**Reply by YamatoYamm** [reply, marked as answer]
I didn't know that exists disscutions about Neuron. Thank you !

**Reply by ieQu1** [reply]
No problem. I notified neuron team about your question.

**Reply by YamatoYamm** [reply]
I have resolved the issue with starting MQTT on Neuron. If you could provide a contact or if you are aware of any specific details about the Neuron platform (I believe someone from the EMQX team would likely have knowledge of these specifics), that would be greatly appreciated. I am working on a project based on OPC UA and MQTT, and Neuron, along with EMQX and MQTTX, seems to be a perfect fit for my project requirements. However, my question is whether Neuron supports or allows for the integration of OPC UA with MQTT, specifically having MQTT operate at the OSI layer 4 while OPC UA operates at a higher layer (5-7). To be more specific, does Neuron perform protocol conversion between OPC UA and MQTT, or is it possible to use OPC UA and MQTT on different OSI layers?

Could you assist with this matter or connect me with someone who might have insights? The challenge lies in the fact that there are some articles on this topic, but it is not explicitly mentioned whether such integration is feasible or not.

Thank you for your help.

**Reply by ieQu1** [reply]
Hello,

EMQX and Neuron are developed and maintained by different teams. It's best to contact Neuron team directly through the dedicated channels, since we, EMQX developers, have a rather limited knowledge of Neuron product. Best we can do is to forward the question.

**Reply by YamatoYamm** [reply]
If you happen to know exactly where it might be possible to reach out to the developers of Neuron to get an answer, please redirect the question elsewhere. 

Thank you !
